File: LeetCode150/Python/Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=len(nums)
        m=nums[0]
        i=1
        for j in range(1,n):
            if(nums[j]==m):
                i+=1
            else:
                i-=1
            
            if(i==0):
                i=1
                m=nums[j]
            
        return m

File: LeetCode150/C++/Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        int m=nums[0];
        int j=1;
        for(int i=1;i<n;i++){

            if(nums[i]==m){
                j++;
            }
            else{
                j--;
            }
            if(j==0){
                j=1;
                m=nums[i];
            }
        }
        return m;
    }
};
